primefaces/src/main/java/org/primefaces/model/JpaLazyDataModel.java

```diff
@@ -366,13 +366,18 @@ protected Object convertToType(Object value, Class valueType) {
                     + "; Create a JSF Converter for it or overwrite Object convertToType(String value, Class<?> valueType)!");
         }
 
-        // first convert the object to string
-        String stringValue = sourceConverter == null
-                ? value.toString()
-                : sourceConverter.getAsString(context, UIComponent.getCurrentComponent(context), value);
-
-        // now convert the string to the required target
-        return targetConverter.getAsObject(context, UIComponent.getCurrentComponent(context), stringValue);
+        try {
+            // first convert the object to string
+            String stringValue = sourceConverter == null
+                    ? value.toString()
+                    : sourceConverter.getAsString(context, UIComponent.getCurrentComponent(context), value);
+
+            // now convert the string to the required target
+            return targetConverter.getAsObject(context, UIComponent.getCurrentComponent(context), stringValue);
+        }
+        catch (Exception e) {
```

Comment:
So this should probably change to ELException?

Reply:
I can change it to ELException. Should I also include IllegalArgumentException, which NumberFormatException extends (among others that could come into play here)?